ext/session/session.c

```diff
@@ -3211,6 +3211,7 @@ static int php_session_rfc1867_callback(unsigned int event, void *event_data, vo
 				if (PS(rfc1867_cleanup)) {
 					php_session_rfc1867_cleanup(progress);
 				} else {
+					SEPARATE_ARRAY(&progress->data);
```

Comment:
If this patch triggers assertions without that change, then I don't think it's separate; If we have to revert one we have to revert the other, they might as well be together imo.